sharding-core/sharding-core-optimize/src/main/java/org/apache/shardingsphere/core/optimize/sharding/segment/select/item/SelectItems.java

```diff
@@ -115,9 +120,18 @@ public boolean isUnqualifiedShorthandItem() {
     public List<String> getColumnLabels() {
         List<String> result = new ArrayList<>(items.size());
         for (SelectItem each : items) {
-            // TODO read * from metadata
             if (!(each instanceof ShorthandSelectItem)) {
                 result.add(each.getColumnLabel());
+            } else {
+                // get tableMetaData from ownerTableMetaDataMap by owner
+                // add tableMetaData's all columnLabels
+                if (((ShorthandSelectItem) each).getOwner().isPresent()) {
```

Comment:
Can you consider about move this logic to `SelectItemsEngine`?

Reply:
Hello, In my opinion, `SelectItemsEngine` should find all related table and table's columns, and `SelectItems` should final exact table and table's columns.
Such as in `SELECT t_order.* FROM t_order, t_order_item WHERE t_order.order_id = t_order_item.order_id`. 
`SelectItemsEngine` find table `t_order`, `t_order_item` and `t_order`'s columns, `t_order_item`'s columns, but `SelectItems` should return `t_order` and `t_order`'s columns.

Reply:
fine